### ISSUE TITLE
[Accessibility]: Branch/Tag/Commit + Repositories selector: Modals have no `title`

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35478]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34417, https://github.com/sourcegraph/sourcegraph/issues/34416

### Problem description

<img width="451" alt="image" src="https://user-images.githubusercontent.com/9516420/168599075-710c6d08-bc4f-41a0-9ee2-62f30a4a5a90.png">

### Expected behavior

We should use `aria-labelledby` to link the modal with the trigger element.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
